### PR TITLE
Add helper methods to compare risks

### PR DESCRIPTION
**Pull request by jayy04**
### Changelist
- Add helper methods to compare risks for future safety heap PRs. Note that this only compares the TNC/MR ratio of the two risk objects, and does not include other constraints (e.g. margin requirements can not increase when subaccount is already under-collateralized) that are used in state transition validations. 

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new method to compare risks between two accounts, improving accuracy by handling rounding errors.

- **Tests**
  - Added tests to ensure the new risk comparison method works correctly with different scenarios.

### CODE REVIEW

**Review by BrendanChou — APPROVED** [review]

File: protocol/lib/margin/risk_test.go [hunk 1]
```diff
@@ -220,3 +220,93 @@ func TestRisk_IsLiquidatable(t *testing.T) {
 		})
 	}
 }
+
+func TestRisk_Cmp(t *testing.T) {
```
Comment (BrendanChou) [inline protocol/lib/margin/risk_test.go:224]: need tests with zero MMR

File: protocol/lib/margin/risk.go [hunk 2]
```diff
@@ -53,6 +53,19 @@ func (a *Risk) IsLiquidatable() bool {
...
 }
 
+// Cmp compares the risks of two accounts.
+// Returns -1 if a is less risky than b, 0 if they are equally risky, and 1 if a is more risky than b.
+
+// Note that here we are effectively checking that
+// `a.NetCollateral / a.MaintenanceMargin >= b.NetCollateral / b.MaintenanceMargin`.
+// However, to avoid rounding errors, we factor this as
+// `a.NetCollateral * b.MaintenanceMargin >= b.NetCollateral * a.MaintenanceMargin`.
+func (a *Risk) Cmp(b Risk) int {
```
Comment (BrendanChou) [inline protocol/lib/margin/risk.go:63]: - consider if `Cmp` is the clearest name as opposed to `CmpRisk` or `CmpSafety` (would be the inverse)
- also consider if we need strict ordering at all (?) like if 2/1 should `Cmp==0` to 4/2

**Review by BrendanChou — CHANGES_REQUESTED** [review]

File: protocol/lib/margin/risk_test.go (hunk 1 above)
Comment (jayy04) [reply protocol/lib/margin/risk_test.go:224]: Added!

File: protocol/lib/margin/risk.go (hunk 2 above)
Comment (jayy04) [reply protocol/lib/margin/risk.go:63]: Think `Cmp` is fine if we want to make it a method on `Risk`. CmpRisk / CmpSafety sounds like a library function that takes in two `Risk` objects. Let me know if you have a preference here. 

> also consider if we need strict ordering at all (?) like if 2/1 should Cmp==0 to 4/2
Added strict ordering. Tie break using NC if a.MMR = b.MMR = 0, otherwise, tie break using MMR. Added corresponding unit tests 🙏

**Review by BrendanChou — APPROVED** [review]
